### ISSUE TITLE
Issue when installing on Raspberry Pi 1

### ISSUE THREAD

**Issue by vialax** [andrei-tatar/node-red-contrib-smartnora#29]
Hi,

I'm having an issue without any idea how to start trying to resolve it when installing smart nora on a raspberry pi 1.

Here are the logs :

2021-05-03T21:06:24.279Z npm install --no-audit --no-update-notifier --no-fund --save --save-prefix=~ --production node-red-contrib-smartnora@1.0.5
2021-05-03T21:07:52.751Z [err] npm ERR! Unexpected end of JSON input while parsing near '...689.tgz","fileCount":'
2021-05-03T21:07:53.657Z [err] 
2021-05-03T21:07:53.657Z [err] npm ERR! A complete log of this run can be found in:
2021-05-03T21:07:53.657Z [err] npm ERR!     /root/.npm/_logs/2021-05-03T21_07_52_692Z-debug.log
2021-05-03T21:07:54.262Z rc=1

Had no problem on the rpi3, is the rpi1 too old to host the node red server with nora ?

Thanks !

**Comment by andrei-tatar** [comment]
@vialax Hi,
What version of node.js and node-red are you running?

**Comment by vialax** [comment]
Hello,

I am using node.js v12.22.1 and node-red v1.3.4

I don't have the debug log anymore, but cleaning the cache and installing through CLI didn't work either.

I will try to get you the log.

**Comment by FireWizard52** [comment]
Hi, @vialax 

It looks that you try to install the modules as root from the CLI. I have no idea how many modules you already have installed as root. But this is not correct. 
You can either install extra nodes (modules) from the palette or from the command line.

If you want to do it from the command line, go to your node red folder. On a Raspberry Pi this is the ~/.node-red.
This is a hidden file (mind the dot) in your home directory (normaly /home/pi)

Follow the procedure to re-install the node modules

Colin wrote:

> In fact, looking back at the start of the thread, it seems somehow your nodes have been messed up. You might be best to rename ~/.node-red/node_modules to something else and then (in the .node-red folder) run
npm install
That will re-install all the manually installed nodes again. Then, if it goes ok, restart node-red. Some nodes can take a long time to install on an old pi so don't worry if it sits for ten minutes apparently doing nothing.

Check that the file package.json exists in the .node-red folder.

After installation restart your node-red

Regards

**Comment by vialax** [comment]
So I did what you said.

7 May 21:50:52 - [info] Waiting for missing types to be registered:
7 May 21:50:52 - [info]  - noraf-config
7 May 21:51:35 - [info] Installing module: node-red-contrib-smartnora,
version: 1.0.5
7 May 21:56:47 - [warn] Installation of module node-red-contrib-smartnora
failed:
7 May 21:56:48 - [warn] ------------------------------------------
7 May 21:56:48 - [warn] npm ERR! code EINTEGRITY
npm ERR!
sha512-0RXEPVODLDYfAvt3wJaxXnDKFaO29OFCMpQ0s5rVjvYKs5ijpzf/FYu78i10HVYoDbjh8ZaZT+EVoxUNZiFq1w==
integrity checksum failed when using sha512: wanted
sha512-0RXEPVODLDYfAvt3wJaxXnDKFaO29OFCMpQ0s5rVjvYKs5ijpzf/FYu78i10HVYoDbjh8ZaZT+EVoxUNZiFq1w==
but got
sha512-cJ+yIboEihmgg/w7Fw5Mi8qD7pZx350RBq9iNfgWZ5HMDjJ7sQOzyP2Mm/SKVvsZGFkR9RlQx3uYfTP8qcAlgQ==.
(1661328 bytes)

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-05-07T20_56_41_906Z-debug.log

It seems that I have two issues, the noraf-config one, and the other one.

Any idea ?

Maybe it will be simpler to reinstall the OS and node red.



Le ven. 7 mai 2021 à 21:48, FireWizard52 ***@***.***> a
écrit :

**Comment by FireWizard52** [comment]
Hello @vialax,

> So I did what you said.

No, it looks that you only tried to install node-red-contrib-smartnora.

I didn't say that, at least it was not my intention to say that. So I might be a little unclear.

What I intended to say was that in the ~/.node-red folder you should rename your folder node_modules to something else.
And then in that directory execute the command npm install.

This should reinstall all the modules in correct folder.

If that is successful, you can restart node-red and then install node-red-contrib-smartnora, either from the palette or from the cli, but if you want to install from the cli, be sure you are in the directory ~/.node-red.

> Maybe it will be simpler to reinstall the OS and node red.

I don't know. It might depend on what other applications has been installed on that Raspberry Pi.

If it is only the OS and Node-Red, I would do that.

Regards

**Comment by vialax** [comment]
That's what i did, but didn't get any luckier.

So fresh install of the OS this mornin, updated/upgraded and installed
nodered flawlessly using the bash script.

Then I trie to install smart nora, here is the console output :

2021-05-09T11:14:35.250Z npm install --no-audit --no-update-notifier
--no-fund --save --save-prefix=~ --production
***@***.***
2021-05-09T11:24:00.789Z [err] npm ERR! Unexpected end of JSON input while
parsing near '..."rollup-plugin-replac'
2021-05-09T11:24:01.035Z [err]
2021-05-09T11:24:01.035Z [err] npm ERR! A complete log of this run can be
found in:
2021-05-09T11:24:01.035Z [err] npm ERR!
/home/pi/.npm/_logs/2021-05-09T11_24_00_390Z-debug.log
2021-05-09T11:24:01.261Z rc=1

Hereafter the last lines of the debug log :

119 verbose stack SyntaxError: Unexpected end of JSON input while parsing
near '..."rollup-plugin-replac'
119 verbose stack     at JSON.parse (<anonymous>)
119 verbose stack     at parseJson
(/usr/lib/node_modules/npm/node_modules/json-parse-better-errors/index.js:7:17)
119 verbose stack     at
/usr/lib/node_modules/npm/node_modules/node-fetch-npm/src/body.js:96:50
119 verbose stack     at runMicrotasks (<anonymous>)
119 verbose stack     at processTicksAndRejections
(internal/process/task_queues.js:97:5)
120 verbose cwd /home/pi/.node-red
121 verbose Linux 5.4.83+
122 verbose argv "/usr/bin/node" "/usr/bin/npm" "install" "--no-audit"
"--no-update-notifier" "--no-fund" "--save" "--save-prefix=~"
"--production" ***@***.***"
123 verbose node v12.22.1
124 verbose npm  v6.14.12
125 error Unexpected end of JSON input while parsing near
'..."rollup-plugin-replac'
126 verbose exit [ 1, true ]

Le sam. 8 mai 2021 à 13:53, FireWizard52 ***@***.***> a
écrit :

**Comment by FireWizard52** [comment]
Hello @vialax 

> So fresh install of the OS this mornin, updated/upgraded and installed
nodered flawlessly using the bash script.

In this case you should have in your home directory (/home/pi) a hidden directory, called .node-red.
In this directory you will also find another directory, called node_modules.
If you enter this directory you will find node-red-contrib-smartnora and that is the only place where this module should be installed.

So far, what the end result should be.

"Googling" for the error:

>  error Unexpected end of JSON input while parsing near .......

points to a cache issue.

As earlier indicated by @andrei-tatar, execute: `npm cache clean --force`

Then enter your .node-red directory and execute: `npm install node-red-contrib-smartnora`

Regards

**Comment by vialax** [comment]
Did the cache clean command, started the install again, got this error now:

***@***.***:~/.node-red $ npm install node-red-contrib-smartnora
npm ERR! code FETCH_ERROR
npm ERR! errno FETCH_ERROR
npm ERR! invalid json response body at
***@***.***%2ffirestore reason: Unexpected end of
JSON input

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-05-09T13_29_07_592Z-debug.log

So basically the same error.

I'll try to reinstall the OS and install nodered through aptitude.

Le dim. 9 mai 2021 à 14:30, FireWizard52 ***@***.***> a
écrit :

**Comment by vialax** [comment]
Fresh installed, tried through palette manager, CLI, still the same error :

pi@raspberrypi:~/.node-red $ npm install --no-audit --no-update-notifier --no-fund --save --save-prefix=~ --production node-red-contrib-smartnora@1.0.5
npm ERR! code FETCH_ERROR
npm ERR! errno FETCH_ERROR
npm ERR! invalid json response body at https://registry.npmjs.org/@firebase%2fauth reason: Unexpected end of JSON input

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-05-09T17_44_17_824Z-debug.log

Couldn't it be an issue on the https://registry.npmjs.org side ?

**Comment by FireWizard52** [comment]
Hello @vialax.

I searched for theerror message and I found:

https://stackabuse.com/npm-error-failed-to-fetch-from-registry-when-installing-module/

Can you check?

And also this:

https://stackoverflow.com/questions/58154616/npm-err-fetcherror-request-to-http-registry-npmjs-org-webpack-dev-server-fai

Regards

**Comment by andrei-tatar** [comment]
@vialax I would also check the card you're using on the pi, or try a different one. Judging on the various errors you're getting I think it's corrupting the files.

**Comment by vialax** [comment]
Yeah you're right, this will be the next thing I test, the card must be the
one I received with the first Pi..

I'll let you know.

Le lun. 10 mai 2021 à 09:11, Andrei Tatar ***@***.***> a
écrit :

**Comment by vialax** [comment]
So, fresh install on another flash card, still no luck :

10 May 19:48:11 - [info] Installing module: node-red-contrib-smartnora, version: 1.0.5
10 May 19:54:41 - [warn] Installation of module node-red-contrib-smartnora failed:
10 May 19:54:41 - [warn] ------------------------------------------
10 May 19:54:41 - [warn] npm ERR! Unexpected end of JSON input while parsing near '...on":"1.8.1-canary.f0f'
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-05-10T18_54_40_557Z-debug.log
10 May 19:54:41 - [warn] ------------------------------------------
Error: Install failed
    at /usr/lib/node_modules/node-red/node_modules/@node-red/registry/lib/installer.js:214:23
10 May 19:54:41 - [error] Error: Install failed
Error: Install failed
    at /usr/lib/node_modules/node-red/node_modules/@node-red/registry/lib/installer.js:214:23
10 May 19:54:42 - [error] Error: Install failed

Then another try :

10 May 20:33:45 - [info] Installing module: node-red-contrib-smartnora, version: 1.0.5
10 May 20:36:32 - [warn] Installation of module node-red-contrib-smartnora failed:
10 May 20:36:32 - [warn] ------------------------------------------
10 May 20:36:32 - [warn] npm ERR! Unexpected end of JSON input while parsing near '...-mocha":"1.3.0","karm'
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-05-10T19_36_31_366Z-debug.log
10 May 20:36:32 - [warn] ------------------------------------------
Error: Install failed
    at /usr/lib/node_modules/node-red/node_modules/@node-red/registry/lib/installer.js:214:23
10 May 20:36:32 - [error] Error: Install failed
Error: Install failed
    at /usr/lib/node_modules/node-red/node_modules/@node-red/registry/lib/installer.js:214:23
10 May 20:36:32 - [error] Error: Install failed

I tried to install the old nora palette, it went flawlessly :

10 May 19:55:35 - [info] Installing module: node-red-contrib-nora, version: 0.0.34
10 May 19:57:40 - [info] Installed module: node-red-contrib-nora
10 May 19:57:49 - [info] Added node types:
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-blinds
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-config
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-scene
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-speaker
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-switch
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-thermostat
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-outlet
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-light
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-garage
10 May 19:57:49 - [info]  - node-red-contrib-nora:nora-lock

Still no way that the issue is server side ? Are your installations working without issues ?

**Comment by FireWizard52** [comment]
@vialax 

No idea, what is wrong with your install. I had some issue, but they were not related to Smart Nora.
When it was solved Smart Nora 1.0.3 installed without any problem.
Later on, it has been upgraded to 1.0.4 and later to 1.0.5. No problems.

**Comment by vialax** [comment]
I'm still trying to install, got always the same kind of issue, but on
different files.

I paste you the logs in case it helps :

***@***.***:~/.node-red $ npm install --no-audit --no-update-notifier
--no-fund --save --save-prefix=~ --production
***@***.***
npm ERR! Unexpected end of JSON input while parsing near
'...es":{"typescript":"3.'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-05-10T20_36_20_038Z-debug.log
***@***.***:~/.node-red $ npm cache clean --force
npm WARN using --force I sure hope you know what you are doing.
***@***.***:~/.node-red $ npm install --no-audit --no-update-notifier
--no-fund --save --save-prefix=~ --production
***@***.***
npm ERR! Unexpected end of JSON input while parsing near
***@***.***'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2021-05-10T20_44_50_340Z-debug.log

Le lun. 10 mai 2021 à 22:42, FireWizard52 ***@***.***> a
écrit :

**Comment by vialax** [comment]
I will try to install others palettes to see if I get this error too or
just with smart Nora.

Is it worth to try to get in touch with npm Devs ?

Le mar. 11 mai 2021 à 09:34, Andrei Tatar ***@***.***> a
écrit :

**Comment by andrei-tatar** [comment]
@vialax any updates on this issue? Did you manage to find the culprit?

**Comment by vialax** [comment]
Hi,

Didn't have much time to look into it, but changing the power supply didn't
really changed the outcome.

I'll tell you when i'll try new things, but I guess i will stick to my rpi3
to host smart nora.

Le lun. 17 mai 2021 à 09:54, Andrei Tatar ***@***.***> a
écrit :

**Comment by andrei-tatar** [comment]
Closing due to inactivity.
